src/silo/preprocessing/metadata_info.cpp

```diff
@@ -0,0 +1,205 @@
+#include "silo/preprocessing/metadata_info.h"
+
+#include <fmt/format.h>
+#include <spdlog/spdlog.h>
+#include <boost/algorithm/string/join.hpp>
+#include <duckdb.hpp>
+
+#include "silo/config/database_config.h"
+#include "silo/preprocessing/preprocessing_exception.h"
+
+namespace {
+
+std::vector<std::string> extractStringListValue(
+   duckdb::MaterializedQueryResult& result,
+   size_t row,
+   size_t column
+) {
+   std::vector<std::string> return_value;
+   duckdb::Value tmp_value = result.GetValue(column, row);
+   std::vector<duckdb::Value> child_values = duckdb::ListValue::GetChildren(tmp_value);
+   std::transform(
+      child_values.begin(),
+      child_values.end(),
+      std::back_inserter(return_value),
+      [](const duckdb::Value& value) { return value.GetValue<std::string>(); }
+   );
+   return return_value;
+}
+
+std::unordered_map<std::string, std::string> validateFieldsAgainstConfig(
+   const std::unordered_map<std::string, std::string> found_metadata_fields,
+   const silo::config::DatabaseConfig& database_config
+) {
+   std::vector<std::string> config_metadata_fields;
+   std::transform(
+      database_config.schema.metadata.begin(),
+      database_config.schema.metadata.end(),
+      std::back_inserter(config_metadata_fields),
+      [](auto metadata) { return metadata.name; }
+   );
+
+   std::unordered_map<std::string, std::string> validated_metadata_fields;
+   for (const auto& pair : found_metadata_fields) {
+      if (std::find(config_metadata_fields.begin(), config_metadata_fields.end(), pair.first)
+       != config_metadata_fields.end()) {
+         validated_metadata_fields.emplace(pair);
+      }
+   }
+   for (const std::string& name : config_metadata_fields) {
+      if (!validated_metadata_fields.contains(name)) {
+         throw silo::preprocessing::PreprocessingException(fmt::format(
+            "The metadata field {} which is contained in the database config is "
+            "not contained in the input.",
+            name
+         ));
+      }
+   }
```

Comment:
Is this the desired behavior? I think in the backend we allow that metadata fields are not present at all if they are not `required = true`. Either we should allow it here or make the backend more strict (either requiring that the proprocessing pipeline adds it or letting the backend add it).

Reply:
Here I would again argue for explicit `null`

Reply:
Ok, then we need to adapt the backend. @chaoran-chen what do you think? Should the backend do it or should the preprocessing pipeline do it?